snapcraft/commands/names.py

```diff
@@ -138,12 +149,25 @@ def run(self, parsed_args):
         if not snaps:
             emit.message("No registered snaps")
         else:
-            tabulated_snaps = tabulate(
-                snaps,
-                headers=["Name", "Since", "Visibility", "Notes"],
-                tablefmt="plain",
-            )
-            emit.message(tabulated_snaps)
+            headers = ["Name", "Since", "Visibility", "Notes"]
+
+            if parsed_args.format == "table":
+                tabulated_snaps = tabulate(
+                    snaps,
+                    headers=headers,
+                    tablefmt="plain",
+                )
+                emit.message(tabulated_snaps)
+            elif parsed_args.format == "json":
+                json_snaps = {
+                    "snaps": [
```

Comment:
Sorry for undoing my approval, but I just thought a bit more about this and would like a bit of discussion:

Why did you choose to make this an object containing a list of snaps rather than just a list of snaps? (I don't think it's necessarily "wrong," I just want to understand this decision.)

Reply:
@lengau I took inspiration from the `snap metrics` output format. By placing the data in a list within an object, metadata can be easily added in future.